perun-core/src/main/java/cz/metacentrum/perun/core/impl/ResourceAssignmentActivator.java

```diff
@@ -0,0 +1,82 @@
+package cz.metacentrum.perun.core.impl;
+
+import cz.metacentrum.perun.core.api.ExtSourcesManager;
+import cz.metacentrum.perun.core.api.GroupResourceAssignment;
+import cz.metacentrum.perun.core.api.GroupResourceStatus;
+import cz.metacentrum.perun.core.api.PerunClient;
+import cz.metacentrum.perun.core.api.PerunPrincipal;
+import cz.metacentrum.perun.core.api.PerunSession;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.lang.NonNull;
+import org.springframework.scheduling.annotation.Async;
+
+import java.util.List;
+
+/**
+ * Component responsible for activating group-resource assignments in PROCESSING or
+ * FAILED state after Perun startup.
+ *
+ * @author Radoslav Čerhák <r.cerhak@gmail.com>
+ */
+public class ResourceAssignmentActivator implements ApplicationListener<ContextRefreshedEvent> {
```

Comment:
I have also done some digging and found out that this event could be actually called twice on startup. I think we could use the `ApplicationReadyEvent` instead. What do you think? Was there a specific reason why to use the `ContextRefreshedEvent`?

Reply:
I didn't know about `ApplicationReadyEvent` but it is a Spring Boot event, so I don't think it would work. Another option might be invoking the activation using `init-method` but it can't run asynchronously - we would have to inject a bean to the activator component and invoke the async call on the bean from `init-method`.

Reply:
I see, I didn't realize that. I think we can keep it this way and we can test it on the devel instance and see, how it behaves.